modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -391,22 +391,13 @@ else if (value instanceof Date[]) {
 				else if (value instanceof Object[]) {
 					Object[] values = (Object[])value;
 
-					String type = field.getType();
-
-					boolean richText = type.equals(
-						DDMFormFieldTypeConstants.RICH_TEXT);
-
 					for (int i = 0; i < values.length; i++) {
-						if (richText) {
-							String valueString = _getSortableValue(
-								ddmStructure.getDDMFormField(field.getName()),
-								locale, values[i].toString());
+						String valueString = _getSortableValue(
+							ddmStructure.getDDMFormField(field.getName()),
+							locale, values[i].toString());
 
-							sb.append(_htmlParser.extractText(valueString));
-						}
-						else {
-							sb.append(values[i]);
-						}
```

Comment:
@rolandpakai why did you remove this `else`?

Reply:
The _addFieldValue logic is used inted of this if-else